tidy3d/web/core/stub.py

```diff
@@ -0,0 +1,66 @@
+"""Stub class"""
+
+from enum import Enum
+
+from pydantic.v1 import BaseModel
+
+
+class TaskType(str, Enum):
+    FDTD = "FDTD"
+    NORMAL = "NORMAL"
+    MODE_SOLVER = "MODE_SOLVER"
+    HEAT = "HEAT"
+
+
+class Stub(BaseModel):
+
+    def from_file(self, file_path: str) -> "Stub":
```

Comment:
note, if you 
```py
from __future__ import annotations
```
then you can use the class name in the type annotations without quotes
```
-> Stub:
```

Reply:
Also, I think `from_file()` should be a @classmethod as well as an @abstractmethod. This way you can call it directly from the class without making an instance.

```
stub = Stub.from_file(path)
```

Reply:
modified